gui/layers/_markers.py

```diff
@@ -0,0 +1,218 @@
+        self._need_display_update = False
+        self._need_visual_update = False
+
+    @property
+    def marker_coords(self):
+        """ndarray: coordinates of the marker centroids
+        """
+        return self._marker_coords
+
+    @property
+    def data(self):
+        """ndarray: coordinates of the marker centroids
+        """
+        return self._marker_coords
+
+    @data.setter
+    def data(self, data):
+        self._marker_coords = data
+        self.refresh()
+
+    @property
+    def symbol(self):
+        """ str: marker symbol
+        """
+        return self._symbol
+
+    @symbol.setter
+    def symbol(self, symbol):
+        self._symbol = symbol
+        self.refresh()
+
+    @property
+    def size(self):
+        """float, ndarray: size of the marker symbol in px
+        """
+
+        return self._size
+
+    @size.setter
+    def size(self, size):
+
+        if isinstance(size, (float, np.ndarray)):
+            self._size = size
+            self.refresh()
+
+        else:
+            raise TypeError('size should be float or ndarray')
+
+    @property
+    def edge_width(self):
+        """float, None: width of the symbol edge in px
+        """
+
+        return self._edge_width
+
+    @edge_width.setter
+    def edge_width(self, edge_width):
+        self._edge_width = edge_width
+        self.refresh()
+
+    @property
+    def edge_width_rel(self):
+        """float, None: width of the marker edge as a fraction
+            of the marker size.
+
+            vispy docs say: "exactly one edge_width and
+            edge_width_rel must be supplied", but I don't know
+            what that means... -KY
+        """
+
+        return self._edge_width_rel
+
+    @edge_width_rel.setter
+    def edge_width_rel(self, edge_width_rel):
+        self._edge_width_rel = edge_width_rel
+        self.refresh()
+
+    @property
+    def edge_color(self):
+        """Color, ColorArray: the marker edge color
+        """
+
+        return self._edge_color
+
+    @edge_color.setter
+    def edge_color(self, edge_color):
+        self._edge_color = edge_color
+        self.refresh()
+
+    @property
+    def face_color(self):
+        """Color, ColorArray: color of the body of the marker
+        """
+
+        return self._face_color
+
+    @face_color.setter
+    def face_color(self, face_color):
+        self._face_color = face_color
+        self.refresh()
+
+    @property
+    def scaling(self):
+        """bool: if True, marker rescales when zooming
+        """
+
+        return self._scaling
+
+    @scaling.setter
+    def scaling(self, scaling):
+        self._scaling = scaling
+        self.refresh()
+
+    def _get_shape(self):
+
+        return np.max(self.marker_coords, axis=0) + 1
+
+    def _update(self):
+        """Update the underlying visual.
+        """
+        if self._need_display_update:
+            self._need_display_update = False
+
+            self.viewer._child_layer_changed = True
```

Comment:
This should actually only be called when information pertaining to the shape/dimensions change. You should separate this from the properties that change how `set_data` is called.

Reply:
Ah okay, this makes sense. I'll move it to the data setter method (the only place we change total size of the layer).